shared/domain/src/commonMain/kotlin/network/bisq/mobile/client/user_profile/ClientUserProfileServiceFacade.kt

```diff
@@ -27,58 +25,41 @@ class ClientUserProfileServiceFacade(
     }
 
     override suspend fun generateKeyPair(result: (String, String, PlatformImage?) -> Unit) {
-        try {
-            val ts = Clock.System.now().toEpochMilliseconds()
-            val preparedData = apiGateway.requestPreparedData()
-            createSimulatedDelay(Clock.System.now().toEpochMilliseconds() - ts)
-            val pubKeyHash: ByteArray = preparedData.id.hexToByteArray()
-            val powSolution = preparedData.proofOfWork.solution
-            val image: PlatformImage? = clientCatHashService.getImage(
-                pubKeyHash,
-                powSolution,
-                0,
-                120
-            )
+        val ts = Clock.System.now().toEpochMilliseconds()
+        val preparedData = apiGateway.requestPreparedData()
+        createSimulatedDelay(Clock.System.now().toEpochMilliseconds() - ts)
+        val pubKeyHash: ByteArray = preparedData.id.hexToByteArray()
+        val powSolution = preparedData.proofOfWork.solution
+        val image: PlatformImage? = clientCatHashService.getImage(
+            pubKeyHash,
+            powSolution,
+            0,
+            120
+        )
 
-            result(preparedData.id, preparedData.nym, image)
-            this.preparedData = preparedData
-        } catch (e: Exception) {
-            log.e("generateKeyPair failed", e)
-        }
+        result(preparedData.id, preparedData.nym, image)
+        this.preparedData = preparedData
     }
 
     override suspend fun createAndPublishNewUserProfile(nickName: String) {
         preparedData?.let { preparedData ->
-            try {
-                val response: UserProfileResponse =
-                    apiGateway.createAndPublishNewUserProfile(
-                        nickName,
-                        preparedData
-                    )
-                this.preparedData = null
-                log.i { "Call to createAndPublishNewUserProfile successful. userProfileId = ${response.userProfileId}" }
-            } catch (e: Exception) {
-                log.e("createAndPublishNewUserProfile failed", e)
-            }
+            val response: UserProfileResponse =
+                apiGateway.createAndPublishNewUserProfile(
+                    nickName,
+                    preparedData
+                )
+            this.preparedData = null
+            log.i { "Call to createAndPublishNewUserProfile successful. userProfileId = ${response.userProfileId}" }
         }
     }
 
     override suspend fun getUserIdentityIds(): List<String> {
-        return try {
-            apiGateway.getUserIdentityIds()
-        } catch (e: Exception) {
-            log.e("getUserIdentityIds failed", e)
-            emptyList()
-        }
+        return apiGateway.getUserIdentityIds()
     }
 
     override suspend fun applySelectedUserProfile(result: (String?, String?, String?) -> Unit) {
```

Comment:
You mean instead to return a wrapper object for the data? Or to not return a data at all in suspend functions?

Reply:
Sorry if it was not clear @HenrikJannsen . Is totally ok to return results, what I mean is instead of passing in a callback function parameter as it is right now:

 `applySelectedUserProfile(result: (String?, String?, String?) -> Unit)`
 
 to avoid that call back function and return the values so that the code can grab it like this
 
 `val (value1, value2, value3) = applySelectedUserProfile()`
 
 does that make sense?